source/extensions/listener_managers/listener_manager/listener_manager_impl.cc

```diff
@@ -409,6 +409,11 @@ bool ListenerManagerImpl::addOrUpdateListener(const envoy::config::listener::v3:
     }
   }
 
+  if (!config.has_internal_listener() && !config.has_api_listener() && !config.has_address()) {
```

Comment:
nit, you needn't check the api listener again, the above already checked.

Reply:
Updated.